src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApApTPC.java

```diff
@@ -267,6 +267,10 @@ protected void buildTxPowerMapForBand(String band, Map<String, Map<String, Integ
 			}
 			JsonArray radioStatuses = allStatuses.get(serialNumber).getAsJsonArray();
 			int currentTxPower = getCurrentTxPower(radioStatuses, band);
+			if (currentTxPower < 0) {
+				// this AP is not on the band of interest
+				continue;
+			}
```

Comment:
Similar to my comment on PR #51 , I prefer you add an explicit check for band instead of inferring it from a seemingly unrelated function. Just add a function like:
```java
// in UCentralUtils
public static String getRadioBand(int index, JsonArray radioStatuses) { ... }
```

Reply:
Discussed this elsewhere - as we can't correlate the two structures, ignore this comment.

Another question though, why are we reading tx power from device config/status instead of state/statistics?

Reply:
There is no good reason I think. I think this is the way it already was. In `MeasurementBasedApClientTPC` it reads it from state.

Reply:
I think if we want to make this "consistent" it should be a separate PR

Reply:
Would you like to include this in your PR or my PR? I'm also good with a separate one.

Reply:
Ok. Since you will be on vacation, I can take care of it, and publish on PR after both our current PRs are merged.

Reply:
Ok. I'll let you (and Jeffrey as needed) coordinate it, thanks!